### ISSUE TITLE
External pipeline unnecessarily running a build

### ISSUE THREAD

**Issue by joshRpowell** [contentful/contentful_middleman#138]
Hi @dlitvakb,

I'm running into a similar issue to this one: https://github.com/middleman/middleman/issues/1916

It popped up with a Netlify build using `middleman contentful --rebuild`.  I wasn't having any issues until Netlify updated their build recently.

The external pipeline is always kicked off when the plugin is activated.  So the `rebuild` runs both build? `true` and `false`: 

```
 activate :external_pipeline,
          name: :webpack,
          command: build? ? 'yarn run build' : 'yarn run start',
          source: '.tmp/dist',
          latency: 1
```

`yarn run start` triggers `webpack --watch` and leaves an open process causing the build to fail.  

Not sure what the best approach is here.  Looking to kick-start the conversation.

**Comment by dlitvakb** [comment]
Hey @joshRpowell,

I don't know how to help you with this, as this is a Netlify related issue, and not related to how `contentful_middleman` works.

Can you open a support ticket on their system?

Cheers

**Comment by joshRpowell** [comment]
@dlitvakb is this a Nelify issue if it occurs locally?

**Comment by dlitvakb** [comment]
I guess not, but, how is this related to the Contentful integration?

I'm asking out of curiosity, not to disregard your question.

**Comment by joshRpowell** [comment]
Learning something new every day...

I went through my notes && logs dating back to the first build on Netlify.  This issue appeared from the very beginning.  Netlify's solution was to run

```
middleman contentful --rebuild && middleman build
```

We were getting the build error, but the second command `middleman build` masked the underlying problem. 

With a fresh head, I decided to take a closer look at the config blocks to identify the difference between the middleman and middleman contentful builds.  I found this in the Middleman Docs: https://github.com/middleman/middleman/blob/7bd64121da3847bb171c956bf717266c14e91f4b/middleman-cli/bin/middleman#L50

Then there was Thomas' comment: https://github.com/middleman/middleman/issues/1916#issuecomment-219766458

Where I noticed this setting: https://github.com/middleman/middleman/blob/7bd64121da3847bb171c956bf717266c14e91f4b/middleman-cli/bin/middleman#L50

It looks like this is the culprit:
```
config[:environment] = (ENV['MM_ENV'] || ENV['RACK_ENV'] || 'development').to_sym
```

Running `middleman contentful --rebuild` would run on the default `development` env.

Setting `MM_ENV=production` solved the problem.
